File: Program/python_and_ruby_101/Mixin/2.rb
module Multiply
  def multiply()
    return @v1 * @v2
  end
end
module Divide
  def divide()
    return @v1 * @v2
  end
end
class Cal
  include Multiply, Divide
  def initialize(v1,v2)
    #생성자 : 인스턴스 만들어질 때 반드시 실행되어야 하는 내용
    p v1, v2
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1+@v2
  end
  def subtract()
    return @v1-@v2
  end
end

c = Cal.new(100,10)
p c.add()
p c.multiply()
p c.divide()

# c2 = CalDivide.new(10,10)
# c2 = c1.add()
# c2 = c1.multiply()

File: Program/python_and_ruby_101/Multiple_inheritance/2.py
class CalMultiply():
    def multiply(self):
        return self.v1*self.v2

class CalDivide():
    def divide(self) :
        return self.v1 / self.v2

class Cal(CalMultiply, CalDivide) :
    #생성자
    def __init__(self, v1, v2):
        if isinstance(v1, int):
            self.v1 = v1
        if isinstance(v2, int):
            self.v2 = v2
    def add(self):
        return self.v1 + self.v2
    def subtract(self):
        return self.v1 - self.v2
    def setV1(self,v):
        if isinstance(v, int):
            self.v1 = v
    def getV1(self) :
        return self.v1


c1 = Cal(10,10)
print(c1.multiply())
print(c1.divide())
